src/test/kotlin/pt/up/fe/ni/website/backend/controller/EventControllerTest.kt

```diff
@@ -151,6 +270,340 @@ internal class EventControllerTest @Autowired constructor(
                 @Test
                 fun `should be a Date`() = validationTester.isDate()
             }
+
+            @Nested
+            @DisplayName("endDate")
+            @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+            inner class EndDateValidation {
+                @BeforeAll
+                fun setParam() {
+                    validationTester.param = "endDate"
+                }
+
+                @Test
+                fun `should be a Date`() = validationTester.isDate()
+            }
+
+            @Nested
+            @DisplayName("url")
+            @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+            inner class UrlValidation {
+                @BeforeAll
+                fun setParam() {
+                    validationTester.param = "url"
+                }
+
+                @Test
+                fun `should be a URL`() = validationTester.isUrl()
+            }
+
+            @Nested
+            @DisplayName("location")
+            @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+            inner class LocationValidation {
+                @BeforeAll
+                fun setParam() {
+                    validationTester.param = "location"
+                }
+
+                @Test
+                @DisplayName("size should be between ${Constants.Location.minSize} and ${Constants.Location.maxSize}()")
+                fun size() =
+                    validationTester.hasSizeBetween(Constants.Location.minSize, Constants.Location.maxSize)
+            }
+
+            @Nested
+            @DisplayName("category")
+            @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+            inner class CategoryValidation {
+                @BeforeAll
+                fun setParam() {
+                    validationTester.param = "category"
+                }
+
+                @Test
+                @DisplayName("size should be between ${Constants.Category.minSize} and ${Constants.Category.maxSize}()")
+                fun size() =
+                    validationTester.hasSizeBetween(Constants.Category.minSize, Constants.Category.maxSize)
+            }
+
+            @Nested
+            @DisplayName("thumbnailPath")
+            @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+            inner class ThumbnailPathValidation {
+                @BeforeAll
+                fun setParam() {
+                    validationTester.param = "thumbnailPath"
+                }
+
+                @Test
+                fun `should be required`() = validationTester.isRequired()
+
+                @Test
+                fun `should be a URL`() = validationTester.isUrl()
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("DELETE /events/{eventId}")
+    inner class DeleteEvent {
+        @BeforeEach
+        fun addEvent() {
+            repository.save(testEvent)
+        }
```

Comment:
Note that this will create an event for every single test we make in the future. We could have some logic checking if it already exists but this should work fine in this endpoint